components/camel-lra/src/main/java/org/apache/camel/service/lra/LRAUrlBuilder.java

```diff
@@ -91,18 +91,12 @@ public LRAUrlBuilder options(Map<String, ?> options) {
 
     public LRAUrlBuilder query(String key, Object value) {
         LRAUrlBuilder copy = copy();
-        try {
-            key = URLEncoder.encode(toNonnullString(key), StandardCharsets.UTF_8.name());
-            value = URLEncoder.encode(toNonnullString(value), StandardCharsets.UTF_8.name());
-            if (copy.query.isEmpty()) {
-                copy.query += "?";
-            } else {
-                copy.query += "&";
-            }
-            copy.query += key + "=" + value;
-        } catch (UnsupportedEncodingException e) {
-            throw new IllegalStateException(e);
+        if (copy.query.isEmpty()) {
+            copy.query += "?";
+        } else {
+            copy.query += "&";
         }
+        copy.query += key + "=" + value;
```

Comment:
```suggestion
        copy.query += toNonnullString(key) + "=" + toNonnullString(value);
```

Reply:
@DFiedler please take a look this suggestion.

Reply:
hi @zhfeng .. i followed your suggestion and adjusted the code